Add HomeKit support for slat tilting

mst opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Proposed change
<!-- 
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
This change adds support to control the slat angle (e.g. venetian blinds) via Apple Home through the HomeKit integration.

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box! 
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request: 

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

mst committed:
Add HomeKit support for slat tilting

bdraco commented on homeassistant/components/homekit/type_covers.py:
```diff
@@ -135,6 +174,21 @@ def update_state(self, new_state):
+                current_tilt - self._homekit_target_tilt < 1
+            ):
+                self.char_target_tilt.set_value(current_tilt)
+                self._homekit_target = None
```
```suggestion
                self._homekit_target_tilt = None
```

I'm not sure this is right.

mst commented on homeassistant/components/homekit/type_covers.py:
```diff
@@ -135,6 +174,21 @@ def update_state(self, new_state):
+                current_tilt - self._homekit_target_tilt < 1
+            ):
+                self.char_target_tilt.set_value(current_tilt)
+                self._homekit_target = None
```
Good catch, @bdraco! I poorly copy-pasted from the positioning code... 🙁

Tbh, I don't understand the reason why this conditional block is (still) needed in the first place. Couldn't this just be?
```suggestion
            self.char_target_tilt.set_value(current_tilt)
```

Is this to avoid syncing back the target_position if initiated by HomeKit? Because you already made sure that this is taken care of in [pyhap.accessory_driver:44](https://github.com/ikalchev/HAP-python/blob/06f73da41ba3a0cdf7f72857f7cd68dc6115667d/pyhap/accessory_driver.py#L454). If so, we could get rid of the `_homekit_target*` instance attributes altogether.

Other than that, the `abs(diff)` piece may make sense if it is to "snap" to the values in the end.

bdraco replied:
Its been in there since @cdce8p added the original support.  It looks like its doing this to normalize the cover position since all covers aren't going to support the precision that homekit is so the state will appear out of sync

mst committed:
Reset tilt-specific attribute, not position attribute

Co-Authored-By: J. Nick Koston <nick@koston.org>

mst commented on homeassistant/components/homekit/type_covers.py:
```diff
@@ -135,6 +174,21 @@ def update_state(self, new_state):
+                current_tilt - self._homekit_target_tilt < 1
+            ):
+                self.char_target_tilt.set_value(current_tilt)
+                self._homekit_target_tilt = None
```
@bdraco, based  on [your comment](https://github.com/home-assistant/core/pull/33388#discussion_r399852059
), this should be `< 6` as well. 
I assume it is for when the device reported back a state that is only _close_ to the one that HomeKit requested (because of different precision), and we're "correcting" HomeKit so that it doesn't appear out of sync.

```suggestion
            # We have to assume that the device has worse precision than
            # HomeKit. If it reports back a state that is only _close_ to HK's
            # requested state, we'll "fix" what HomeKit requested.
            DEVICE_PRECISION_LEEWAY = 6
            if self._homekit_target_tilt is None or abs(
                current_tilt - self._homekit_target_tilt < DEVICE_PRECISION_LEEWAY
            ):
                self.char_target_tilt.set_value(current_tilt)
                self._homekit_target_tilt = None
```

bdraco replied:
LGTM.  Would you please move the constant into const.py and use it for the position as well.

mst committed:
Add explanation why we fix HomeKit's targets

We have to assume that the device has worse precision than HomeKit. If it
reports back a state that is only _close_ to HK's requested state, we'll
"fix" what HomeKit requested so that it won't appear out of sync.

bdraco commented:
LGTM
